examples/version-migration/live-schema-upgrade/src/modelVersion2/diceCounter.ts

```diff
@@ -56,8 +56,8 @@ export class DiceCounter extends DataObject implements IDiceCounter {
 		if (this._counter === undefined) {
 			// Get the existing counter if we didn't initialize it.
 			const sharedCounterHandle =
-				this.root.get<IFluidHandle<ISharedCounter>>(sharedCounterKey);
-			assert(sharedCounterHandle !== undefined, "sharedCounterHandle should be defined");
+				this.root.get<IFluidHandle<ISharedCounter>>(sharedCounterKey) ??
+				fail("sharedCounterHandle should be defined");
```

Comment:
changing this to use fail is a subjective stylistic choice: the old assert would have been fine, but I find this pattern a little cleaner (and its more efficient if you want to do formatting in the message string), and I wanted to include an example use of fail in this PR.

Reply:
I like this pattern too.